### PR TITLE
applications: asset_tracker_v2: Add new "nod" run-time configuration

### PR DESCRIPTION
**Pull request by simensrostad**
Add support for a new run-time configuration that enables
sample requests for GNSS and neighbor cell data to be enabled/disabled
at run-time.

At JSON protocol level, the configuration is called "nod" which is short
for No Data. This is a list that includes data types that are not
going to be included by the application module when sample requests are
sent out to other modules. This enables the user to configure via the
web interface the source of location-related data that is sampled on
the device. If the list is empty both GNSS and neighbor cell data
is sampled.

In the application, this list translates to boolean variables that are
used to toggle the inclusion of the corresponding data type in the
`APP_EVT_DATA_GET` event.

### CODE REVIEW

File: applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h [hunk 1]
```diff
@@ -95,6 +106,8 @@ struct cloud_data_cfg {
 	int movement_timeout;
 	/** Accelerometer trigger threshold value in m/s2. */
 	double accelerometer_threshold;
+	/** Variable used to govern what data types that are requested by the application. */
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h:109]:
```suggestion
	/** Variable used to govern what data types are requested by the application. */
```

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c [hunk 2]
```diff
@@ -887,6 +923,30 @@ void json_common_config_get(cJSON *parent, struct cloud_data_cfg *data)
...
+			if (strcmp(item->valuestring, CONFIG_NO_DATA_LIST_NEIGHBOR_CELL) == 0) {
+				ncell_found = true;
+			}
+		}
+
+		/* If a supported entry is present in the no data list we set the corresponding flag
+		 * to false.
+		 */
+		data->no_data.gnss = (gnss_found) ? false : true;
+		data->no_data.neighbor_cell = (ncell_found) ? false : true;
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c:948]:
```suggestion
		data->no_data.gnss = (gnss_found) ? true : false;
		data->no_data.neighbor_cell = (ncell_found) ? true : false;
```

File: applications/asset_tracker_v2/src/modules/Kconfig.data_module [hunk 3]
```diff
@@ -135,6 +135,22 @@ config DATA_GPS_TIMEOUT_SECONDS
 	help
 	  Timeout for acquiring a GPS fix during sampling of data.
 
+config DATA_GNSS_REQUEST_ON_SAMPLE
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/Kconfig.data_module:138]:
```suggestion
config DATA_SAMPLE_GNSS_DEFAULT
```

File: applications/asset_tracker_v2/src/modules/Kconfig.data_module [hunk 4]
```diff
@@ -135,6 +135,22 @@ config DATA_GPS_TIMEOUT_SECONDS
...
+	bool "Include GNSS in sample requests"
+	default y
+	help
+	  If this configuration is enabled the application will by default include GNSS data in
+	  sample requests sent to other modules. This configuration can be overwritten by changing
+	  the applications real-time configuration via the web interface.
+
+config DATA_NEIGHBOR_CELLS_REQUEST_ON_SAMPLE
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/Kconfig.data_module:146]:
```suggestion
config DATA_SAMPLE_NEIGHBOR_CELLS_DEFAULT
```

File: applications/asset_tracker_v2/src/modules/Kconfig.data_module [hunk 5]
```diff
@@ -135,6 +135,22 @@ config DATA_GPS_TIMEOUT_SECONDS
 	help
 	  Timeout for acquiring a GPS fix during sampling of data.
 
+config DATA_GNSS_REQUEST_ON_SAMPLE
+	bool "Include GNSS in sample requests"
+	default y
+	help
+	  If this configuration is enabled the application will by default include GNSS data in
+	  sample requests sent to other modules. This configuration can be overwritten by changing
+	  the applications real-time configuration via the web interface.
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/Kconfig.data_module:144]:
```suggestion
	  the application's real-time configuration using the cloud-side state.
```

File: applications/asset_tracker_v2/src/modules/Kconfig.data_module [hunk 6]
```diff
@@ -135,6 +135,22 @@ config DATA_GPS_TIMEOUT_SECONDS
...
+
+config DATA_NEIGHBOR_CELLS_REQUEST_ON_SAMPLE
+	bool "Include neighbor cells in sample requests"
+	default y
+	help
+	  If this configuration is enabled the application will by default include neighbor cell
+	  data in sample requests sent to other modules. This configuration can be overwritten by
+	  changing the applications real-time configuration via the web interface.
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/Kconfig.data_module:152]:
```suggestion
	  changing the application's real-time configuration using the cloud-side state.
```

**Review by coderbyheart — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c [hunk 7]
```diff
@@ -840,6 +840,43 @@ int json_common_config_add(cJSON *parent, struct cloud_data_cfg *data, const cha
 		values_added = true;
 	}
 
+	if (data->nod_list_fresh) {
+		cJSON *nod_list = cJSON_CreateArray();
```
Comment (jtguggedal) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c:844]: Missing NULL check of this one?

Comment (simensrostad) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c:844]: Jepp. fixed.

**Comment by simensrostad** [comment]
@umapraseeda @jtguggedal Squashed the doc-review fixes and added the missing null check.

**Review by jtguggedal — APPROVED** [review]
Thanks

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**